cadc-log/src/main/java/ca/nrc/cadc/log/LogControlServlet.java

```diff
@@ -359,29 +371,173 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     /**
      * Check for proper group membership.
      */
-    private void authorize(HttpServletRequest request, boolean readOnly) throws AccessControlException, TransientException {
-        if (authorizerClassName != null && groupAuthorizer == null) {
-            throw new RuntimeException("CONFIG: group authorizer was configured but failed to load: " + authorizerClassName);
+    private void authorize(HttpServletRequest request, boolean readOnly)
+        throws AccessControlException, TransientException {
+
+        // Track exceptions thrown
+        AccessControlException acException = null;
+
+        // Get the logControl properties if they exist.
+        PropertiesReader propertiesReader = getLogControlProperties();
+
+        // first check if request user matches authorized config file users
+        Set<Principal> authorizedUsers = getAuthorizedUserPrincipals(propertiesReader);
+        if (authorizedUsers.isEmpty()) {
+            logger.debug("Authorized users not configured");
+        } else {
+            try {
```

Comment:
By throwing an exception as a response to what is essentially a question, this is using a try/catch to do logic. It would be cleaner for doUserCheck to return true if the caller is in the list and the code read something like:
`if (doUserCheck(caller, authorizedUsers)) {
   return;
}`
Below, once a doGroupCheck(caller, groupUris) returns true you can also return immediately.

Then try the Authorizer (which you have).

The last line of this method should be 
`throw new AccessControlException(...)`

Reply:
Updated the auth checking methods to return a boolean and removed the try clauses.